### PR TITLE
Support for host definition templates

### PR DESCRIPTION
**Pull request by janneojala**

### CODE REVIEW

File: templates/etc/nagios3/conf.d/generic-host_nagios2.cfg.j2 [hunk 1]
```diff
@@ -19,3 +19,33 @@ define host{
 	contact_groups                  admins
 	register                        0       ; DONT REGISTER THIS DEFINITION - ITS NOT A REAL HOST, JUST A TEMPLATE!
 }
+
+{% if nagios_server_host_templates is defined %}
+{% for host_template in nagios_server_host_templates %}
```
Comment (tersmitten) [inline templates/etc/nagios3/conf.d/generic-host_nagios2.cfg.j2]: Can you change this to:

```
{% for host_template in nagios_server_host_templates | default([]) %}
```

Comment (tersmitten) [reply templates/etc/nagios3/conf.d/generic-host_nagios2.cfg.j2]: And remove the `if`

File: templates/etc/nagios3/conf.d/generic-host_nagios2.cfg.j2 [hunk 2]
```diff
@@ -19,3 +19,33 @@ define host{
...
+        retain_nonstatus_information    {{ host_template.retain_nonstatus_information | default(1) }}       ; Retain non-status information across program restarts
+        check_command                   {{ host_template.check_command | default('check-host-alive') }}
+        max_check_attempts              {{ host_template.max_check_attempts | default(10) }}
+        notification_interval           {{ host_template.notification_interval | default(0) }}
+        notification_period             {{ host_template.notification_period | default('24x7') }}
+        notification_options            {{ host_template.notification_options | default('d,u,r') }}
+        contact_groups                  {{ host_template.contact_groups | default('admins') }}
+        register                        {{ host_template.register | default(0) }}       ; DONT REGISTER THIS DEFINITION - ITS NOT A REAL HOST, JUST A TEMPLATE!
+ {% if host_template.variables is defined %}
+ {% for host_template_variable in host_template.variables %}
```
Comment (tersmitten) [inline templates/etc/nagios3/conf.d/generic-host_nagios2.cfg.j2]: Can you change this:

```
{% for host_template_variable in host_template.variables | default([]) %}
```

Comment (tersmitten) [reply templates/etc/nagios3/conf.d/generic-host_nagios2.cfg.j2]: And remove the `if`

File: templates/etc/nagios3/conf.d/hostname_nagios2.cfg.j2 [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # {{ ansible_managed }}
 
 define host{
-	use                     generic-host            ; Name of host template to use
+	use                     {{ item.host_template | default('generic-host') }}            ; Name of host template to use
```
Comment (tersmitten) [inline templates/etc/nagios3/conf.d/hostname_nagios2.cfg.j2:4]: Can you add this variable to the `README`.

**Comment by tersmitten** [comment]
Looks like a real nice addition! I'll try to test this as soon as possible.

**Comment by janneojala** [comment]
Test template would be nice, I'd like to learn proper way to performs tests.

**Comment by tersmitten** [comment]
Some daemons support something [like this](https://github.com/Oefenweb/ansible-haproxy/blob/master/tasks/main.yml#L77), but I don't know if nagios supports it and if it works with multiple configuration files

**Comment by janneojala** [comment]
Invoking nagios -v nagios.cfg does the validation, how to implement this with travis?

**Comment by tersmitten** [comment]
You don't do it in travis, you do it were you [write the templates](https://github.com/Oefenweb/ansible-nagios-server/blob/master/tasks/configure.yml)

**Comment by janneojala** [comment]
Yes of course

**Comment by janneojala** [comment]
As you said, adding "validate" -keyword while templating jinja2 -files will do. However nagios can't validate single config files, nagios.cfg only.

**Pull request merged**